### DISCUSSION TITLE
Get ListItemAllFields after added file to a folder

### DISCUSSION THREAD

**Discussion by pINKmUNster** [pnp/pnpcore#380] [Q&A] [answered]
Hello, 

I'm trying to add list item field while adding a file. 
I saw that i can use the ListItemAllFields property but when I try to add a new value in it my code failed. 

I have this error
 ```
PnP.Core.ClientError
PnP.Core.ClientException: Please first fetch this list from SharePoint before doing operations on it
   at PnP.Core.Model.SharePoint.ListItem.BuildUpdateApiCallAsync(PnPContext context)
   at PnP.Core.Model.SharePoint.ListItem.BaseUpdate(Func`2 fromJsonCasting, Action`1 postMappingJson)
   at PnP.Core.Model.BaseDataModel`1.UpdateAsync()
```

This my dummy code : 

```
var addedFile = await folder.Files.AddAsync(fileName, stream);
addedFile.ListItemAllFields["aaa"] = "aaaaa";
await addedFile.ListItemAllFields.UpdateAsync();
```

Do I need to recatch the file and load the ListItemAllFields or do i'm missing something? 

Thanks  😃

**Comment by jansenbe** [comment]
You'll need to load ListItemAllFields that property is not expanded when doing an add. Add this line after the AddAsync call:

```csharp
await addedFile.ListItemAllFields.LoadAsync();
```

**Comment by pINKmUNster** [comment, marked as answer]
It's working as expected. 
Thank a lot
